feat(xo-lite): Storage Repositories

schulz-alexander opened the pull request:
### Description

Storage Repositories in main navigation
- SR dashboard with general information and usage
- VDI list

![image](https://github.com/vatesfr/xen-orchestra/assets/5782951/64bfcc77-0450-46a0-94ca-6e8b5e773d48)

![image](https://github.com/vatesfr/xen-orchestra/assets/5782951/59416713-8a23-4aa4-bece-6e44f8fb36e4)


### Checklist

- Commit
  - Title follows [commit conventions](https://bit.ly/commit-conventions)
  - Reference the relevant issue (`Fixes #007`, `See xoa-support#42`, `See https://...`)
  - If bug fix, add `Introduced by`
- Changelog
  - If visible by XOA users, add changelog entry
  - Update "Packages to release" in `CHANGELOG.unreleased.md`
- PR
  - If UI changes, add screenshots
  - If not finished or not tested, open as _Draft_

schulz-alexander committed:
feat(xo-lite): Storage Repositories

schulz-alexander committed:
feat(xo-lite): Storage Repositories

schulz-alexander committed:
feat(xo-lite): Storage Repository: translation fix for general section

schulz-alexander committed:
feat(xo-lite): Storage Repository: translation fix for general section

olivierlambert commented:
Thanks a lot for your contribution @borzel :+1: 

@clemencebx will come back to you here with some screenshots on how we planned to display the storage in the treeview (it's a bit different IIRC). Stay tuned!

clemencebx commented:
Hello @borzel ! That's already really great :)
Here is where we are leading for the storage treeview. The idea will be to separate VMs, storages, and networks so the architecture of the treeview will work well for both XO Lite and XO 6.

![XO6 - VMs Treeview](https://github.com/vatesfr/xen-orchestra/assets/82316090/f4e31218-0611-41e6-8e4c-84bfdb9eab72) ![XO6 - Storage Treeview](https://github.com/vatesfr/xen-orchestra/assets/82316090/708f8799-36ca-4fe5-8b78-388a2c34d771)

schulz-alexander commented:
Thanks :-)

Should I try to implement this design (at least for VMs and Storages)? Or is someone else internally on it?

schulz-alexander commented:
Second question: What is "big_boss" in your design? A new layer on top of pools?

clemencebx commented:
"big_boss" represents the XOA site as some users have multiple XOA, sometimes based on location, somtimes based on usage etc... They need a way to differentiate them, so each site has its own name. For the sake of the design, I randomly chose to call the site big_boss :)

schulz-alexander commented:
Progress happens :-)

![grafik](https://github.com/vatesfr/xen-orchestra/assets/5782951/ba996bb6-3ba7-42d1-85f0-89576f38bfe8)

julien-f committed:
feat(xo-server/signin): remember me with external providers

It works the same as password signin.

julien-f committed:
feat(xo-server/signin): uniformize sign in buttons

Make sign in with password button the same as external providers.

olivierlambert commented:
Just to keep you posted @borzel : it takes longer than usual because we are also preparing the common base for XO 6 and XO Lite. This requires a lot of big changes in the code base, as per:
* https://github.com/vatesfr/xen-orchestra/pull/7302
* https://github.com/vatesfr/xen-orchestra/pull/7292

Be assured we'll come back to you when this is done :+1:

schulz-alexander committed:
Merge branch 'vatesfr:master' into xo-lite-sr

ByScripts committed:
feat(lite): upgrade deps + root eslint config (#7292)

pdonias committed:
fix(xo-web/new-vm): isDiskTemplate → _isDiskTemplate (#7311)

julien-f committed:
chore: update dev deps

julien-f committed:
fix(xo-server-backup-reports): require Node>=15

Introduced by 2af74008b

Due to using AggregateError.

OlivierFL committed:
fix(xo-web/metadata-restore-modal): add check to hide pool select when restoring xo config backup (#7287)

See https://xcp-ng.org/forum/topic/8130/xo-configbackup-restore

MrGrymReaper committed:
feat(lite): introduce PWA manifest (#7291)

This implements the initial PWA manifest for XO Lite. I requested this several
months to a year ago, so decided to do it myself in the end.

julien-f committed:
feat(gitignore): ignore .tap/

julien-f committed:
feat(xo-server): find plugins sequentially

This provides a deterministic order.

In case of duplicate plugins (with the same name), the first found plugin takes precedence.

julien-f committed:
feat(xo-server): move plugin lookup paths to config

julien-f committed:
feat(xo-server): load plugins from mono-repo

Contrary to 3e3ce543a85e7d471804f59aff13876abe83fe86, which was reverted,
this implentation properly handle duplicates.

julien-f committed:
feat(self-signed): readCert utility (#7282)

Expired certificates are not automatically detected, which is not a big deal for user certificates because they can still be used and it's their responsibility to update them.

But automatic certificates must be regenerated in that case which was not the case until now.

This commit unifies certificate/key reading, checking and generation for both xo-server and xo-proxy.

fbeauchamp committed:
fix(fuse-vhd): potential race condition in mount/unmount (#7312)

The code was not properly waiting mount/unmount to be done.

OlivierFL committed:
fix(xo-web/modal): close modal when navigating to another url (#7301)

ddelnano committed:
feat(xo-server): add xenStoreData to XO VM objects (#7316)

The initial support added in #7055 to support terraform resource support doesn't provide read access to a VM's state.

Since Tterraform's model requires read and write access to the resource it's managing, this PR implements the missing piece for terra-farm/terraform-provider-xenorchestra#261.

schulz-alexander committed:
current state

schulz-alexander committed:
current state

ByScripts commented:
Just a quick update to let you know that we're not ignoring this PR.

We're still working on the foundation of the future XO Lite and XO 6, with lots of changes involved, new tree logic and components, integration of the new "Design System", and so on.

I'll take a look as soon as the core is ready to work with and I have more availability.

Thanks again for your participation in the project.
